### ISSUE TITLE
Error when creating postgresql using ibm_database resource

### ISSUE THREAD

**Issue by mariouhrin** [IBM-Cloud/terraform-provider-ibm#879]
### Terraform Version
v0.19.0

### Affected Resource(s)
- ibm_database

### Terraform Configuration Files
```hcl
provider "ibm" {
  region           = "${var.region}"
  ibmcloud_api_key = "${var.ibmcloud_api_key}"
  resource_group   = "${var.resource_group}"
}

data "ibm_resource_group" "group" {
  name = "${var.resource_group}"
}

resource "ibm_database" "postgresql" {
  name              = "${var.db_pg_name}"
  plan              = "standard"
  location          = "${var.region}"
  service           = "databases-for-postgresql"
  resource_group_id = "${data.ibm_resource_group.group.id}"

  service_endpoints            = "private"
  adminpassword                = "${var.db_pg_password}"
  members_memory_allocation_mb = 2048
  members_disk_allocation_mb   = 10240
}
```

### Expected Behavior
Should output connection string without error

### Actual Behavior
```
Error: Error applying plan:

1 error occurred:
        * module.create_db_postgresql.ibm_database.postgresql: 1 error occurred:
        * ibm_database.postgresql: Error getting user connection string for user (admin): Error getting database user connection string via ICD API: Request failed with status code: 404, ServerErrorResponse: {"errors":"not_found"}
```

### Steps to Reproduce
1. `terraform init`
2. `terraform plan`
3. `terraform apply -auto-approve`

**Comment by hkantare** [comment]
As per doc we need to set the provider region 
https://ibm-cloud.github.io/tf-ibm-docs/v0.19.0/r/database.html
```
Configuration of an ICD resource requires that the region parameter is set for the IBM provider in the provider.tf to be the same as the target ICD location/region. If not specified it will default to us-south. A terraform apply will fail if the ICD location is set differently. 
```

**Comment by mariouhrin** [comment]
The region for provider and database is the same.

The problem is with the private service endpoint because the default (public) works with 0.19.0

That terraform script above is capable to create the database but always fails in the end with this problem:
```
Error getting user connection string for user (admin): Error getting database user connection string via ICD API: Request failed with status code: 404, ServerErrorResponse: {"errors":"not_found"}
```
 
I would like to create unit tests for this script and therefore it shouldn't fail

Anyway using the version 0.17.6 with below script works:

```hcl
resource "ibm_resource_instance" "db_postgresql" {
  name              = "${var.db_pg_name}"
  plan              = "standard"
  location          = "${var.region}"
  service           = "databases-for-postgresql"
  resource_group_id = "${var.resource_group_id}"

  parameters = {
    service_endpoints            = "private"
    members_memory_allocation_mb = 2048
    members_disk_allocation_mb   = 10240
  }

  timeouts {
    create = "60m"
    update = "60m"
    delete = "60m"
  }
}
```

**Comment by hkantare** [comment]
Fix is available in master.

**Comment by mariouhrin** [comment]
@hkantare thanks for fixing the issue

**Comment by hkantare** [comment]
Fixed in v0.20.0 https://github.com/IBM-Cloud/terraform-provider-ibm/releases/tag/v0.20.0
